src/import/ensureConfiguration.ts

```diff
@@ -0,0 +1,596 @@
+ * @returns {string|undefined} path to the config file or undefined if it couldn't find or create one
+ */
+export function ensureConfiguration(root: string, allowFromScratch = false): string {
+  const configPath = [path.join(root, 'config.yaml'), path.join(root, 'config.yml')].find(
+    fs.existsSync
+  );
+  if (configPath) {
+    // The config already exists, so return it
+    logger.info(`Using configuration file: ${path.resolve(configPath)}`);
+    return configPath;
+  }
+
+  // The config doesn't exist, so generate one
+  return generateConfiguration(root, allowFromScratch);
+}
+
+/**
+ * Creates a new config.yaml file using available other configs (package.json, ig.ini, package-list.json, menu.xml)
+ * @param root - the root path of the FSH Tank
+ * @param allowFromScratch - create a config file from scratch even if package.json isn't found
+ * @returns {string|undefined} path to the config file or undefined if it couldn't find or create one
+ */
+function generateConfiguration(root: string, allowFromScratch: boolean): string {
+  // Generate the config from available ig.ini, package.json, and package-list.json
+  const igIni = getIgIni(root);
+  const packageJSON = getPackageJSON(root);
+  const packageList = getPackageList(root);
+
+  // The config is "from scratch" if there is no package.json to base it on.
+  // We know there wasn't a packageJSON if the returned packageJSON is the default one (by reference)
+  const isFromScratch = packageJSON === DEFAULT_PACKAGE_JSON;
+  if (isFromScratch && !allowFromScratch) {
+    return;
+  }
+
+  // Create the new YAML document
+  const configPath = path.join(root, 'config.yaml');
+  const doc = new Document();
+  const contents = new CommentableYAMLMap();
+  // @ts-ignore See: https://github.com/eemeli/yaml/issues/156
+  doc.contents = contents;
+
+  // Set the comments for the document header. Comments differ depending on which
+  // files were used in generation (if any).
+  let docComments: string;
+  if (isFromScratch) {
+    docComments =
+      'This SUSHI config.yaml file is provided as an example. You must modify its values to ' +
+      'accurately reflect the details of your project.';
+  } else {
+    docComments =
+      'This SUSHI config.yaml file was generated from the following configuration file(s):\n';
+    if (packageJSON !== DEFAULT_PACKAGE_JSON) {
+      docComments += '* package.json\n';
+    }
+    if (igIni !== DEFAULT_IG_INI) {
+      docComments += `* ${path.join('ig-data', 'ig.ini')}\n`;
+    }
+    if (packageList !== DEFAULT_PACKAGE_LIST) {
+      docComments += `* ${path.join('ig-data', 'package-list.json')}\n`;
+    }
+    docComments +=
+      'Please review this new config.yaml file and, make edits as necessary. This file should be considered the ' +
+      'new gold source configuration for SUSHI. Since this is the new gold source, you should delete the files ' +
+      'listed above since they are now unnecessary and redundant with this file.';
```

Comment:
You're right on both counts.  Will fix it.  The bit about deleting files was written before we decided to support the files in place.

Reply:
Based on guidance from Mark, I slimmed down the comments a LOT.  Still, I put in ALL CAPS saying to review and edit the file.  And now deleting files is a little less important since by default, if we find an existing file, we just use it.  They should still delete package.json, which I now note in the (much shorter) comment -- but it was weird when I made that part ALL CAPS too -- so it's in normal caps.